src/core/server/saved_objects/migrationsv2/integration_tests/migration_from_older_v1.test.ts

```diff
@@ -159,136 +162,50 @@ describe('migration v2', () => {
-    });
+    const response = body[migratedIndex];
 
-    it('migrates the documents to the highest version', async () => {
-      const expectedVersions = getExpectedVersionPerType();
-      const res = await esClient.search({
-        index: migratedIndex,
-        body: {
-          sort: ['_doc'],
-        },
-        size: 10000,
-      });
-      const allDocuments = res.body.hits.hits as SavedObjectsRawDoc[];
-      allDocuments.forEach((doc) => {
-        assertMigrationVersion(doc, expectedVersions);
-      });
-    });
+    expect(response).toBeDefined();
+    expect(Object.keys(response.aliases!).sort()).toEqual(['.kibana', `.kibana_${kibanaVersion}`]);
   });
 
-  describe('migrating from the same Kibana version that used v1 migrations', () => {
-    const originalIndex = `.kibana_1`; // v1 migrations index
-    const migratedIndex = `.kibana_${kibanaVersion}_001`;
-
-    beforeAll(async () => {
-      await removeLogFile();
-      await startServers({
-        oss: false,
-        dataArchive: Path.join(
-          __dirname,
-          'archives',
-          '8.0.0_v1_migrations_sample_data_saved_objects.zip'
-        ),
-      });
-    });
-
-    afterAll(async () => {
-      await stopServers();
-    });
-
-    it('creates the new index and the correct aliases', async () => {
-      const { body } = await esClient.indices.get(
-        {
-          index: migratedIndex,
-        },
-        { ignore: [404] }
-      );
-      const response = body[migratedIndex];
-
-      expect(response).toBeDefined();
-      expect(Object.keys(response.aliases!).sort()).toEqual([
-        '.kibana',
-        `.kibana_${kibanaVersion}`,
-      ]);
-    });
-
-    it('copies the documents from the previous index to the new one', async () => {
-      // original assertion on document count comparison (how atteched are we to this assertion?)
-      const migratedIndexResponse = await esClient.count({
-        index: migratedIndex,
-      });
-      const oldIndexResponse = await esClient.count({
-        index: originalIndex,
-      });
-
-      // Use a >= comparison since once Kibana has started it might create new
-      // documents like telemetry tasks
-      expect(migratedIndexResponse.body.count).toBeGreaterThanOrEqual(oldIndexResponse.body.count);
+  it('copies all the document of the previous index to the new one', async () => {
+    const originalDocs = await fetchDocuments(esClient, originalIndex);
+    const migratedDocs = await fetchDocuments(esClient, migratedIndex);
+    expect(assertMigratedDocuments(migratedDocs, originalDocs));
+  });
 
-      // new assertion against a document array comparison
-      const originalDocs = await fetchDocuments(esClient, originalIndex);
-      const migratedDocs = await fetchDocuments(esClient, migratedIndex);
-      expect(assertMigratedDocuments(migratedDocs, originalDocs));
+  it('migrates the documents to the highest version', async () => {
```

Comment:
@rudolf  what changes fixed flakiness in the test?

Reply:
The original failure in https://github.com/elastic/kibana/issues/87968 was due to a "Network Error" with an unhelpful stack trace and the Jenkins Builds no longer exist so it's hard to really understand what went wrong.

So I have done two things:
 1. Split up the test so there's one ES/Kibana per integration test. In other tests I have seen it cause problems when we start Kibana twice from the same test suite. I think this is probably the most likely root cause.
 2. Don't count the documents, but find all documents and compare their _id's instead which has been stable for other tests. I can't quite figure out why, but somehow the count API returns a lower count directly after the migration and if we wait a bit it gives the expected count.

I'm running the flaky test runner to double check that this solves the flakiness

Reply:
no test failures after 42 runs https://kibana-ci.elastic.co/job/kibana+flaky-test-suite-runner/1904/